lib/galaxy/config/__init__.py

```diff
@@ -133,37 +133,36 @@ def _set_config_directories():
                 self.config_dir = os.path.abspath(self.config_dir)
 
             self.data_dir = config_kwargs.get('data_dir')
-            # mutable_config_dir is intentionally not configurable. You can
-            # override individual mutable configs with config options, but they
-            # should be considered Galaxy-controlled data files and will by default
-            # just live in the data dir
             self.sample_config_dir = os.path.join(os.path.dirname(__file__), 'sample')
+            self.managed_config_dir = config_kwargs.get('managed_config_dir')
 
             if running_from_source:
                 if self.config_dir is None:
                     self.config_dir = os.path.join(self.root, 'config')
                 if self.data_dir is None:
                     self.data_dir = os.path.join(self.root, 'database')
-                self.mutable_config_dir = self.config_dir
+                if self.managed_config_dir is None:
```

Comment:
Question: if the admin sets `managed_config_dir` to the empty string `''`, it is neither resolved to an absolute path (even with my suggestion above, which follows what is done for `config_dir`) nor overridden here.

It's not clear what an admin would intend, but I think we should decide how to interpret that.

The same applies for `config_dir`, `data_dir`.

Reply:
I think setting a config option to `''` is a valid use case. That's how an admin would override the schema default if they wanted no value to be set. It will indeed break things if the `config_dir` path is set to `''`, but so would other unreasonable settings. Besides, in this case galaxy would fail fast and loudly: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/__init__.py#L839

So maybe just leave as is?

Otherwise, one way to handle it, I think, would be to change these lines: https://github.com/galaxyproject/galaxy/blob/823526a1fc6c14c868f473f067e68c37ecd5197a/lib/galaxy/config/__init__.py#L140 (6 total) from 
`if self.foo is None:` 
to 
`if not self.foo:`

So, we treat `''` same as "not set", and set it to defaults.
What do you think?

Reply:
Changing to `if not` would also be my preferred solution.

Reply:
OK, can do.